homeassistant/components/switchbee/__init__.py

```diff
@@ -35,30 +22,25 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     central_unit = entry.data[CONF_HOST]
     user = entry.data[CONF_USERNAME]
     password = entry.data[CONF_PASSWORD]
-    devices_map: dict[str, DeviceType] = {s.display: s for s in DeviceType}
-    allowed_devices = [
-        devices_map[device]
-        for device in entry.options.get(CONF_DEVICES, CONF_DEFUALT_ALLOWED)
-    ]
+
     websession = async_get_clientsession(hass, verify_ssl=False)
     api = CentralUnitAPI(central_unit, user, password, websession)
     try:
         await api.connect()
-    except SwitchBeeError:
-        return False
+    except SwitchBeeError as exp:
+        raise PlatformNotReady("Failed to connect to the Central Unit") from exp
```

Comment:
This doesn't work. Use `ConfigEntryNotReady`. Please make sure to test your changes.

Reply:
After changing to `ConfigEntryNotReady ` I simulated this by disconnecting the dev machine from the network, I had to catch `aiohttp.client.ClientConnectorError` as well but now I see a log which says `Config entry '192.168.1.116' for switchbee integration not ready yet: Failed to connect to the Central Unit; Retrying in background`